vars/runGradleCheck.groovy

```diff
@@ -0,0 +1,59 @@
+void call(Map args = [:]) {
+    def lib = library(identifier: 'jenkins@20211123', retriever: legacySCM(scm))
+    def git_repo_url = args.gitRepoUrl ?: 'null'
+    def git_reference = args.gitReference ?: 'null'
+
+    println("Git Repo: ${git_repo_url}")
+    println("Git Reference: ${git_reference}")
+
+    if (git_repo_url.equals('null') || git_reference.equals('null')) {
+        println("No git repo url or git reference to checkout the commit, exit 1")
+        System.exit(1)
+    }
+    else {
+        withCredentials([
+            usernamePassword(credentialsId: "jenkins-gradle-check-s3-aws-credentials", usernameVariable: 'amazon_s3_access_key', passwordVariable: 'amazon_s3_secret_key'),
+            usernamePassword(credentialsId: "jenkins-gradle-check-s3-aws-resources", usernameVariable: 'amazon_s3_base_path', passwordVariable: 'amazon_s3_bucket'),
+            usernamePassword(credentialsId: "jenkins-github-bot-token", usernameVariable: 'GITHUB_BOT_USER', passwordVariable: 'GITHUB_BOT_TOKEN')]) {
+
+            sh """
+
+                set -e
+                set +x
+
+                env | grep JAVA | grep HOME
+
+                echo "Git clone: ${git_repo_url} with ref: ${git_reference}"
+                rm -rf search
+                git clone ${git_repo_url} search
+                cd search/
+                git checkout -f ${git_reference}
+                git rev-parse HEAD
+
+                echo "Stop existing gradledaemon"
+                ./gradlew --stop
+                find ~/.gradle -type f -name "*.lock" -delete
+
+                echo "Check existing dockercontainer"
+                docker ps -a
+                docker stop `docker ps -qa` > /dev/null 2>&1 || echo
+                docker rm `docker ps -qa` > /dev/null 2>&1 || echo
+                echo "Stop existing dockercontainer"
```

Comment:
Is this a precautionary measure? AFAIK each executor will have its own container right?

Reply:
Gradle check have issues with running on containers and has been confirmed with core engineers, it also does not help to run in containers as gradle check will spawn new containers for testing, thus cause a container in container situation which is hard to clean up.

I have synced with @bbarani and we are going to run directly on Agent node, just like fork jenkins setup now.

Therefore it is needed to make sure there is no existing docker container running as gradle check will spin up new containers.

Reply:
Can we link the issue here in the comment: https://github.com/opensearch-project/OpenSearch/issues/3674? Later we can run gradle check in the container